pandas/core/frame.py

```diff
@@ -4715,6 +4714,7 @@ def sort_values(
         inplace=False,
         kind="quicksort",
         na_position="last",
+        key=None,
```

Comment:
type

Reply:
This wasn't typed because of a mypy issue I can't figure out how to fix. The generic.py `sort_values` function has the `by` parameter defaulted to `None`, while `frame.py` doesn't have the default. This causes a mypy issue because the inheritance pattern is kind of screwed up, but leaving key untyped seemingly confuses mypy enough so it doesn't raise a warning. For reference the error is `Signature of "sort_values" incompatible with supertype "NDFrame"`. I'm inclined to `type : ignore` it, but maybe there's a solution?

Reply:
@jreback Also this – I don't see a way around the basic incompatibility of the `Series` and `NDFrame` `sort_values` functions.

Reply:
cc @simonjayhawkins on the typing question.

Reply:
Can you add docs on what is passed to the `key` function? Is it the `Series`, obtained from `self[by]`?

Reply:
Yea this seems tricky to resolve; I think OK as a follow up

Reply:
Actually if you can type `key` and ignore error code specific to inheritance would be best. Can do this as of mypy 0.730

http://mypy-lang.blogspot.com/2019/09/mypy-730-released.html

Reply:
isn't this just a Callable?

Reply:
I added #type: ignore[override] to ignore this. However, the current release of flake8 raises spurious errors for this syntax, so we will have to add `# NOQA` on those two lines until the next `pyflakes` release (it has been fixed in master, see https://github.com/PyCQA/pyflakes/issues/475).